src/Language/PureScript/Linter/Exhaustive.hs

```diff
@@ -223,22 +229,20 @@ missingAlternative env mn ca uncovered
   mcases = missingCases env mn uncovered ca
 
 -- |
--- Main exhaustivity checking function
+-- Main exhaustivity checking function.
 -- Starting with the set `uncovered = { _ }` (nothing covered, one `_` for each function argument),
 -- it partitions that set with the new uncovered cases, until it consumes the whole set of clauses.
 -- Then, returns the uncovered set of case alternatives.
 -- 
 checkExhaustive :: forall m. (MonadWriter MultipleErrors m) => Environment -> ModuleName -> [CaseAlternative] -> m ()
-checkExhaustive env mn cas = makeResult . first nub $ foldl' step ([initial], (pure True, [])) cas
+checkExhaustive env mn cas = makeResult . first (nub . take maxCases) $ foldr step ([initial], (pure True, [])) cas
   where
-  step :: ([[Binder]], (Maybe Bool, [[Binder]])) -> CaseAlternative -> ([[Binder]], (Maybe Bool, [[Binder]]))
-  step (uncovered, (nec, redundant)) ca =
+  step :: CaseAlternative -> ([[Binder]], (Maybe Bool, [[Binder]])) -> ([[Binder]], (Maybe Bool, [[Binder]]))
+  step ca (uncovered, (nec, redundant)) =
     let (missed, pr) = unzip (map (missingAlternative env mn ca) uncovered)
-        cond = or <$> sequenceA pr
-    in (concat missed, (liftA2 (&&) cond nec,
+        cond = or <$> sequence pr
+    in (if fromMaybe True cond then concat missed else uncovered, (liftA2 (&&) cond nec,
```

Comment:
Yes, actually, it's something I came up when writing the library. When we have that `cond` is `Just False`, we know that the case doesn't cover anything. So, it doesn't generate new uncovered cases, i.e., we maintain the previous `uncovered` list.